fuel-vm/src/constraints/reg_key.rs

```diff
@@ -304,39 +368,22 @@ impl<'a> From<WriteRegisters<'a>> for WriteRegistersRef<'a> {
     }
 }
 
-#[test]
-fn can_split() {
-    let mut reg: [Word; VM_REGISTER_COUNT] = std::iter::successors(Some(0), |x| Some(x + 1))
-        .take(VM_REGISTER_COUNT)
-        .collect::<Vec<_>>()
-        .try_into()
-        .unwrap();
-    let expect = reg;
-
-    let (r, w) = split_registers(&mut reg);
-
-    assert_eq!(r.zero, RegMut::<ZERO>(&mut (ZERO as u64)));
-    assert_eq!(r.one, RegMut::<ONE>(&mut (ONE as u64)));
-    assert_eq!(r.of, RegMut::<OF>(&mut (OF as u64)));
-    assert_eq!(r.pc, RegMut::<PC>(&mut (PC as u64)));
-    assert_eq!(r.ssp, RegMut::<SSP>(&mut (SSP as u64)));
-    assert_eq!(r.sp, RegMut::<SP>(&mut (SP as u64)));
-    assert_eq!(r.fp, RegMut::<FP>(&mut (FP as u64)));
-    assert_eq!(r.hp, RegMut::<HP>(&mut (HP as u64)));
-    assert_eq!(r.err, RegMut::<ERR>(&mut (ERR as u64)));
-    assert_eq!(r.ggas, RegMut::<GGAS>(&mut (GGAS as u64)));
-    assert_eq!(r.cgas, RegMut::<CGAS>(&mut (CGAS as u64)));
-    assert_eq!(r.bal, RegMut::<BAL>(&mut (BAL as u64)));
-    assert_eq!(r.is, RegMut::<IS>(&mut (IS as u64)));
-    assert_eq!(r.ret, RegMut::<RET>(&mut (RET as u64)));
-    assert_eq!(r.retl, RegMut::<RETL>(&mut (RETL as u64)));
-    assert_eq!(r.flag, RegMut::<FLAG>(&mut (FLAG as u64)));
-
-    for i in 0..VM_REGISTER_WRITE_COUNT {
-        assert_eq!(w.0[i], i as u64 + 16);
+impl TryFrom<RegisterId> for WriteRegKey {
+    type Error = RuntimeError;
+    fn try_from(ra: RegisterId) -> Result<Self, Self::Error> {
```

Comment:
[nit] does ra mean anything special here? Usually it refers to the first register in an opcode but that doesn't seem relevant in this case.